### PR TITLE
Detect Tuned profile and incorporate into custom profile

### LINKED ISSUE #1: Hardcoded active profile
In your ERB template to create the custom tuned conf, you hardcoded that it was a virtual-guest instead of detecting the active profile and basing it off that.

### PR DESCRIPTION
**Pull request by jhg03a**
This fixes #1 by adding a pair of new custom puppet facts and incorporating them into the custom profile.

### CODE REVIEW

File: templates/tuned.conf.erb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [main]
-include=virtual-guest
+include=<%= scope.lookupvar('::tuned_base_profile') %>
```
Comment (alex-harvey-z3q) [inline templates/tuned.conf.erb:2]: I think this should be better implemented as `<%= @tuned_base_profile %>`.  See doco [here](https://docs.puppet.com/puppet/latest/reference/lang_template_erb.html#variable).

File: manifests/init.pp [hunk 2]
```diff
@@ -47,13 +47,14 @@
       content => template("${module_name}/tuned.conf.erb"),
     }
 
-    exec { 'enable-tuned-profile':
-      command => '/sbin/tuned-adm profile custom',
-      unless  => '/sbin/tuned-adm active | grep -q "custom"',
+    if $::tuned_active_profile != 'custom' {
```
Comment (alexharv074) [inline manifests/init.pp]: I would prefer unless here:

```puppet
unless $::tuned_active_profile == 'custom' {
```

**Comment by alex-harvey-z3q** [comment]
Actually I changed my mind, I'm going to merge your PR and fix up the issues myself as I think that's going to be faster.  Thanks very much for the contribution @jhg03a

**Comment by alex-harvey-z3q** [comment]
Actually I forgot, I can't merge it if the build is broken.  How would you like to proceed here @jhg03a - do you have time to figure out why the tests are failing?  Otherwise I can raise a new PR based on your work.

File: lib/facter/tuned.active.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
+require 'facter'
+
+def get_active_profile
+  active_profile = ""
```
Comment (alex-harvey-z3q) [inline lib/facter/tuned.active.rb]: I prefer consistent use of single & double quoting, and for me use of single quote denotes a simple string and use of double quotes indicates that something is going to be interpolated inside the string.  So I think you can fix this just by replacing all double quotes with single quotes.

**Comment by jhg03a** [comment]
That's weird, I've pushed the fixes to my master and branch.  I can clone them from github.  However, the website and this PR havn't updated.

**Comment by jhg03a** [comment]
Ok, appears github was having issues.  Things are good now.

**Pull request merged**